test/unit/server.js

```diff
@@ -113,4 +143,65 @@ describe("CommunityProductServer", () => {
         assert(server.communities[contract.address])
         assert(!server.communities[contract2.address])
     })
+
+    it("will not fail to start if there is an error playing back a community", async function () {
+        log("Starting CommunityProductServer...")
+        const storeDir = path.join(os.tmpdir(), `communitiesRouter-test1-${+new Date()}`)
+        const config = {
+            tokenAddress,
+            operatorAddress: wallet.address,
+        }
+        const server = new CommunityProductServer(wallet, storeDir, config, log, log)
+        sinon.stub(server, "getStoreFor").callsFake(() => mockStore(startState, initialBlock, log))
+        sinon.stub(server, "getChannelFor").callsFake(() => new MockStreamrChannel("dummy-stream-id"))
+
+        await server.start()
+
+        const contract = await deployTestCommunity(wallet, wallet.address, tokenAddress, 1000, 0)
+        log(`Deployed contract at ${contract.address}`)
+
+        const contract2 = await deployTestCommunity(wallet, wallet.address, tokenAddress, 1000, 0)
+        log(`Deployed contract at ${contract2.address}`)
+
+        await server.communityIsRunning(contract.address)
+        await server.communityIsRunning(contract2.address)
+        log("Communities running")
+
+        await server.stop()
+        await sleep(ganacheBlockIntervalSeconds * 1000)
+
+        // force one community startup to fail when getting channel
+        server.getChannelFor.withArgs(contract.address).callsFake(async function () {
+            throw new Error("expected fail")
+        })
+        await assert.doesNotReject(() => server.start())
```

Comment:
cool test, still lots of setup, but the intervention and assert are short and very readable :)